### PR TITLE
Add more cluster, track and gen particle information

### PR DESCRIPTION
**Pull request by selvaggi**
BEGINRELEASENOTES
- set mass to KL or Photon for neutral clusters respectively
- store position and time of cluster via calorimeterHit
- set first track hit as MC truth track vertex
- added vertex time to MC truth particle

ENDRELEASENOTES

### CODE REVIEW

File: converter/src/DelphesEDM4HepConverter.cc [hunk 1]
```diff
@@ -269,12 +281,23 @@ namespace k4SimDelphes {
...
-      const auto momentum = delphesCand->P4();  // NOTE: assuming massless here!
+      // NOTE: Delphes assumes m=0 for photons and m=0.497611 for neutral hadrons (KL)
+      const auto momentum = delphesCand->P4();
       // TODO: fill this when it is available later, when when we link the references?
       // cand.setCharge(delphesCand->Charge);
       cand.setMomentum({(float)momentum.Px(), (float)momentum.Py(), (float)momentum.Pz()});
       cand.setEnergy(delphesCand->E);
-
+      cand.setMass(delphesCand->P4().M());
+      // NOTE: Particle-Flow Neutral are either photons or K_L in Delphes
```
Comment (tmadlener) [inline converter/src/DelphesEDM4HepConverter.cc:291]: Is there any chance that we get non-neutral candidates here? In principle those should all go to the `Tracks`, right? I.e. `Tower` will always be a neutral?

**Review by tmadlener — APPROVED** [review]
Thanks for taking care of this. Since we already discussed this quite a bit, I would merge this as is (as soon as the nightlies build passes). I just have one clarifying question (to a comment in fact), but that should be fine as well.

File: converter/src/DelphesEDM4HepConverter.cc (hunk 1 above)
Comment (selvaggi) [reply converter/src/DelphesEDM4HepConverter.cc:291]: It is possible for a charged hadron/electron to  fall in the `EFlowNeutralHadron`/`EFlowPhoton` collections and form a `Tower`, but in that case Delphes simply creates a K_L or a Photon.

**Comment by tmadlener** [comment]
Tested this just now against delphes@9bdb6b7d (master from before the weekend) and everything compiles there. The direct comparison tests fail currently, due to some jet constituent issue.

I would merge this now, unless anything is still holding this back and then deal with the jet constituents issue in a separate PR.

**Pull request merged**